examples/gzip/gzip-envoy.yaml

```diff
@@ -0,0 +1,124 @@
+              request_direction_config:
+                common_config:
+                  enabled:
+                    default_value: false
+                    runtime_key: request_compressor_enabled
+              compressor_library:
+                name: text_optimized
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+                  memory_level: 3
+                  window_bits: 10
+          - name: envoy.filters.http.router
+            typed_config: {}
+  - address:
+      socket_address:
+        address: 0.0.0.0
+        port_value: 8002
+    filter_chains:
+    - filters:
+      - name: envoy.filters.network.http_connection_manager
+        typed_config:
+          "@type": type.googleapis.com/envoy.extensions.filters.network.http_connection_manager.v3.HttpConnectionManager
+          codec_type: AUTO
+          stat_prefix: ingress_http
+          route_config:
+            name: local_route
+            virtual_hosts:
+            - name: backend
+              domains:
+              - "*"
+              routes:
+              - match:
+                  prefix: "/stats/prometheus"
+                route:
+                  cluster: envoy-stats
+          http_filters:
+          - name: envoy.filters.http.compressor
+            typed_config:
+              "@type": type.googleapis.com/envoy.extensions.filters.http.compressor.v3.Compressor
+              response_direction_config:
+                common_config:
+                  min_content_length: 100
+                  content_type:
+                    - text/plain
+                disable_on_etag_header: true
+              request_direction_config:
+                common_config:
+                  enabled:
+                    default_value: false
+                    runtime_key: request_compressor_enabled
+              compressor_library:
+                name: text_optimized
+                typed_config:
+                  "@type": type.googleapis.com/envoy.extensions.compression.gzip.compressor.v3.Gzip
+                  memory_level: 3
+                  window_bits: 10
+          - name: envoy.filters.http.router
+            typed_config: {}
+
+  clusters:
+  - name: envoy-stats
+    connect_timeout: 0.25s
+    type: STATIC
+    load_assignment:
+      cluster_name: envoy-stats
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: 127.0.0.1
+                port_value: 8001
+  - name: service
+    connect_timeout: 0.25s
+    type: STRICT_DNS
+    lb_policy: ROUND_ROBIN
+    load_assignment:
+      cluster_name: service
+      endpoints:
+      - lb_endpoints:
+        - endpoint:
+            address:
+              socket_address:
+                address: service
+                port_value: 8080
+admin:
+  address:
+    socket_address:
+      address: 0.0.0.0
+      port_value: 8001
```

Comment:
use port ``10000`` as that is already exposed in the container - also i think use ``10001`` for the compressed stats